### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by pdalpra**
- Introduce base Spec traits, BaseSpec and AkkaSpec
- move ValidationValues up to io.gatling.core
- Proper use of Akka TestKit
- Do not use IO in TcpSenderSpec => no random fail anymore
- Re-implement HTTP tests support with cleaner Netty-based server

### CODE REVIEW

File: gatling-http/src/test/scala/io/gatling/http/integration/HttpIntegrationSpec.scala [hunk 1]
```diff
@@ -15,112 +15,101 @@
...
+                regexCheck("<title>Resource Test</title>"))),
+        protocolCustomizer = _.inferHtmlResources(BlackList(".*/bad_resource.png")))
+
+      session.isFailed shouldBe false
+
+      verifyRequestTo("/resourceTest/index.html")
+      verifyRequestTo("/resourceTest/stylesheet.css")
+      verifyRequestTo("/resourceTest/script.js")
+      verifyRequestTo("/resourceTest/img.png")
+      verifyRequestTo("/bad_resource.png", 0)
```
Comment (slandelle) [inline gatling-http/src/test/scala/io/gatling/http/integration/HttpIntegrationSpec.scala]: should be "/resourceTest/bad_resource.png"

**Comment by slandelle** [comment]
`HttpIntegrationSpec` fails on Travis. :(

**Comment by pdalpra** [comment]
Strange...
It runs fine on my machine... maybe the port selection m

**Comment by slandelle** [comment]
Tests fail immediately on my machine. Investigating.

**Pull request merged**